### PR TITLE
Use FutureOr.

### LINKED ISSUE #26: Implementations of Future.then should be updated to use FutureOr
Classes that implement `Future` should change the signature ot `.then` to be

```dart
then(FutureOr<S> onValue(T x), {Function onError})
```

to ensure that inference will continue to work for calls to `.then` and to get better error checking.

### PR DESCRIPTION
**Pull request by nex3**
Closes #26

### CODE REVIEW

File: lib/src/delegate/future.dart [hunk 1]
```diff
@@ -27,7 +27,7 @@ class DelegatingFuture<T> implements Future<T> {
   Future<T> catchError(Function onError, {bool test(Object error)}) =>
     _future.catchError(onError, test: test);
 
-  Future/*<S>*/ then/*<S>*/(dynamic onValue(T value), {Function onError}) =>
+  Future/*<S>*/ then/*<S>*/(FutureOr/*<S>*/ onValue(T value), {Function onError}) =>
```
Comment (kevmoo) [inline lib/src/delegate/future.dart]: Might as well use full generic method syntax, too – right?

Comment (nex3) [inline lib/src/delegate/future.dart]: Lasse has a separate CL out for that.

Comment (lrhn) [inline lib/src/delegate/future.dart]: I have landed that now, so you can update the one on `FutureOr`.

**Comment by lrhn** [comment]
LGTM

**Pull request merged**